backend/api/cms/news/queries/news_article.py

```diff
@@ -6,7 +6,8 @@
 
 @strawberry.field
 def news_article(hostname: str, slug: str, language: str) -> NewsArticle | None:
-    site = Site.objects.filter(hostname=hostname).first()
+    hostname, port = hostname.split(":") if ":" in hostname else (hostname, 80)
```

Comment:
Do we have the request object somewhere? 
Maybe we can use `Site.find_for_request(request)` https://docs.wagtail.org/en/stable/reference/pages/model_reference.html#wagtail.models.Site.find_for_request ?

Reply:
Unfortunately, I don't believe it would work with our current setup as we have a hacky middleware that forces the Host of the request to be pycon.it (hopefully, one day, we will get rid of it), so it wouldn't be able to match with 2023.pycon.it / 2024.pycon.it

Once we remove the middleware, it could be an option, but it would limit our API as you won't be able to request any of our CMS data anymore unless you pass through a layer that makes sure the API hostname matches the CMS hostname. I am not against it, as in terms of security, this endpoint allows everyone to fetch any page for any site we have in our CMS, which is not ideal